MinecraftClient/ConsoleIO.cs

```diff
@@ -123,29 +123,29 @@ public static void WriteLineFormatted(string str, bool acceptnewlines = false, b
             
             if (!String.IsNullOrEmpty(str))
             {
-                if (!acceptnewlines) 
+                if (displayTimestamp == null)
                 {
-                    output.Append(str.Replace('\n', ' '));
+                    displayTimestamp = EnableTimestamps;
                 }
-                else 
+                if (displayTimestamp.Value)
                 {
-                    output.Append(str);
+                    int hour = DateTime.Now.Hour, minute = DateTime.Now.Minute, second = DateTime.Now.Second;
+                    output.Append(String.Format("{0}:{1}:{2} ", hour.ToString("00"), minute.ToString("00"), second.ToString("00")));
                 }
-                if (displayTimestamp == null) 
+                if (!acceptnewlines)
                 {
-                    displayTimestamp = EnableTimestamps;
+                    str = str.Replace('\n', ' ');
                 }
-                if (displayTimestamp.Value) 
+                if (BasicIO_NoColor)
                 {
-                    int hour = DateTime.Now.Hour, minute = DateTime.Now.Minute, second = DateTime.Now.Second;
-                    output.Append(String.Format("{0}:{1}:{2} ", hour.ToString("00"), minute.ToString("00"), second.ToString("00")));
+                    output.Append(ChatBot.GetVerbatim(str));
+                }
+                else
+                {
+                    output.Append(str);
                 }
                 if (BasicIO)
                 {
-                    if (BasicIO_NoColor)
-                    {
-                        output.Append(ChatBot.GetVerbatim(str));
-                    }
                     Console.WriteLine(output.ToString());
                     return;
```

Comment:
Since `BasicIO` is `true` when `BasicIO_NoColor` is `true`, it's possible to put the if check for `BasicIO_NoColor` back with a little bit of extra code.

```cs
    if (BasicIO)
    {
        if (BasicIO_NoColor)
            output.Append(ChatBot.GetVerbatim(str));
        else 
            output.Append(str);

        Console.WriteLine(output.ToString());
        return;
    }
    else
    {
        output.Append(str);
    }
```

I don't recall why I added a `StringBuilder` here, and it doesn't look to me like it's doing anything useful, so it might be best to just revert the `StringBuilder` changes.